### ISSUE TITLE
parse5 doesn't emit onParseError for some HTML LS tree-correction steps (math-in-head, stray tag/text after body)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3967]
## Summary

Three `nu-only` fixtures share a root cause: HTML LS's tree construction algorithm specifies these markup patterns as `Parse error` (or produce a corrected-but-nonconformant tree), but parse5 v8.0.1 — the parser markuplint is built on — silently performs the tree correction without calling its `onParseError` callback (`_err()`) at the relevant step. Since markuplint's rule layer only ever sees the *corrected* DOM (`packages/@markuplint/ml-core`) and its only token/parse-error-level detection channel is `parse-error` (relayed from parse5's `onParseError`, see `packages/@markuplint/html-parser/src/parser.ts` and the `mirrorsParseErrorCodes` mechanism documented in `packages/@markuplint/rules/CLAUDE.md`), none of these three cases are currently detectable by any markuplint rule — the "wrongness" of the original markup is structurally invisible after parsing.

This is an upstream (parse5) implementation gap, not a markuplint rule-authoring gap: there is no DOM-shape difference for these rules to look at, and the token-level signal markuplint depends on (`onParseError`) never fires.

### 1. `html-math/math-in-head-novalid.html`

```html
<head><meta charset=utf-8><title>...</title><math><mi>x</mi></math></head><body></body>
```

nu-validator: error — `Start tag "body" seen but an element of the same type was already open.` / `Stray end tag "head".`

Spec: [HTML LS §4.2.1 the head element](https://html.spec.whatwg.org/multipage/semantics.html#the-head-element), content model — metadata content only (`base`/`link`/`meta`/`noscript`/`script`/`style`/`template`/`title`); MathML's `<math>` is not metadata content. `packages/@markuplint/html-spec/src/spec.head.jsonc` already encodes this content model (`":model(metadata):not(title)"`) correctly — the rule that checks it (`permitted-contents`) simply never sees a violation because parse5's "in head" insertion-mode "Anything else" step ([HTML LS §13.2.6.4.4](https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inhead)) pops `<head>` and re-parents `<math>` under a synthesized `<body>` *before* markuplint's DOM ever exists. Verified empirically: `parse5.parse()` places `<math>` as a child of `<body>`, not `<head>`, for this input.

### 2. `html/parser/stray-start-tag-novalid.html`

```html
<body>
</body>
<p>stray paragraph</p>
```

nu-validator: error — `Stray start tag "p".`

Spec: [HTML LS §13.2.6.4.17 "after body" insertion mode, "Anything else"](https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-afterbody): "Parse error. Switch the insertion mode to 'in body' and reprocess the token." Verified empirically: parse5 re-parents `<p>` into `<body>` per this exact algorithm step, but its `tokenAfterBody` handler (parse5 v8.0.1 internals) never calls `_err()` there — `parseErrors` comes back empty for this input.

### 3. `html/parser/text-after-body-novalid.html`

```html
<body></body>text
```

nu-validator: error (subType `fatal`) — `Cannot recover after last error. Any further errors will be ignored.` / error — `Non-space character after body.`

Same spec citation and same parse5 code path as #2 (`tokenAfterBody`'s character-token branch) — the text is re-parented into `<body>` without `_err()` firing. (nu's first message is nu's own internal recovery-tracking diagnostic, not an HTML LS parse error in its own right — only the second message maps to spec text.)

## Why this isn't a markuplint rule fix

`packages/@markuplint/rules/CLAUDE.md` documents the `mirrorsParseErrorCodes` mechanism (a rule declares which parse5 error codes it subsumes, so `ml-core` suppresses the duplicate generic `parse-error` violation) — but that mechanism only works when parse5 *does* emit the error. Here it doesn't, for any of the three cases, so there is no signal for a rule (existing or new) to consume. Confirmed via `document.parseErrors.length === 0` for all three fixtures when parsed through `@markuplint/html-parser`.

## Possible directions (none attempted; needs discussion)

1. **Upstream fix in parse5**: file the gap against `inikulin/parse5` (the "in head"/"after body" `_err()` calls are apparently missing for these specific branches) and pick up the fix on the next parse5 bump. Lowest risk, but timeline is out of this project's control.
2. **Independent token-stream scan in `@markuplint/html-parser`**: run a lightweight pre-pass over the raw HTML (or hook parse5's tokenizer output directly, bypassing tree construction) to detect these specific stray-tag-after-body / non-metadata-in-head patterns before parse5's tree correction erases them. Significant new surface area; risks duplicating parse5's own state machine.
3. **Accept as an architectural limitation**: document these fixtures as an intentionally-uncovered case (not `excluded-ids.json` — the spec does forbid the markup, so this isn't nu over-detection — but a documented gap) and revisit if/when parse5 fixes upstream.

## Files

- `packages/@markuplint/html-parser/src/parser.ts`
- `packages/@markuplint/rules/src/permitted-contents/` (fixture #1's would-be rule, currently blocked)
- `node_modules/parse5` (v8.0.1, pinned in `packages/@markuplint/html-parser/package.json`) — the actual gap lives here

## Bench impact

3 fixtures currently `nu-only`. No fix is proposed here yet — this Issue exists to record the root-cause analysis so triage doesn't re-derive it, and to host discussion on which of the three directions above (if any) is worth pursuing.

**Comment by YusukeHirao** [comment]
Duplicate of #3943, which already tracks all 3 of these fixtures (math-in-head, stray-start-tag, text-after-body) with the same root-cause analysis (parse5 not calling `onParseError`/`_err()` for these tree-correction steps). Closing in favor of the existing Issue.
